bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Condition.java

```diff
@@ -22,7 +23,22 @@
  *
  * @author Yordan Mihaylov - Initial Contribution
  */
-public interface Condition extends Module {
+public class Condition extends Module {
+
+    private Set<Connection> connections;
+
+    /**
+     * Constructor of {@link Condition} module object.
+     *
+     * @param id id of the module.
+     * @param typeUID unique module type id.
+     * @param configuration configuration values of the {@link Condition} module.
+     * @param connections set of {@link Connection}s used by this module.
+     */
+    public Condition(String id, String typeUID, Map<String, ?> configuration, Set<Connection> connections) {
```

Comment:
We will need default constructor for GSON. This also applies for Trigger and Action